controllers/suite_test.go

```diff
@@ -55,6 +55,11 @@ func setup() error {
 	// Uncomment the below to run the tests in the old v1 naming mode
 	// envy.Set("AZURE_SECRET_NAMING_VERSION", "1")
 
+	// TODO: Should this just default to true?
```

Comment:
Need to decide on this before merging the PR

Reply:
@babbageclunk do you have thoughts here?

Reply:
Without this on, everyone who has soft-delete on in their KV is potentially going to hit the same problem, right? (And it's not going to affect KVs without soft-delete.) There's versioning so that it's not lossy in the way that a purge would be. How fine-grained are the permissions? Is permission to recover the same as deleting? 
I think defaulting it to true would make sense.

Reply:
Yeah it's not lossy, and KeyVault is actually requiring softdelete to be enabled starting sometime end of this year I think so it will become a common problem.

Reply:
There is a different permission (recover) that is required to perform a recovery. But if we don't have such a permission and we're in this case we have the choice of:

Failing because we can't create the key
Failing because we can't recover the key

So either way... it's going to fail in this situation.

Reply:
Yeah, not much we can do in that case, and the recovery behaviour is much better than purging on delete. I say do it!

Reply:
I don't see a downside - worst case is that we fail in a scenario where we would have failed _anyway_.

The upside is that we cleanly handle a situation without failing, improving resilience.

Reply:
Done, the default is now true